web/src/api/index.ts

```diff
@@ -82,7 +130,7 @@ export async function getListeners(): Promise<Listener[]> {
 
 type Stacklet = components['schemas']['Stacklet'];
 export async function getStacklets(): Promise<Stacklet[]> {
-  const { data } = await client.get('/stacklets', {});
+  const { data } = await client.get('/stacklets', sessionOptions());
```

Comment:
I guess we should pass all required options (like the auth header) when creating the client. This ensures that every API call is done with the required header. This however means to things:

- We might need a separate client which handles auth (without the header included)
- We have to wrap the client in our own logic to provide dynamic session token updating via signals

Reply:
c2600fcf74521bf1eeb9324b03ab401382345397